drivers/i2c/i2c_ll_stm32.c

```diff
@@ -43,26 +43,60 @@ LOG_MODULE_REGISTER(i2c_ll_stm32);
 #else
 #define STM32_I2C_DOMAIN_CLOCK_SUPPORT 0
 #endif
+int i2c_stm32_get_config(const struct device *dev, uint32_t *config)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	struct i2c_config_timing *cfg_timings = (void *)config;
+	I2C_TypeDef *i2c = cfg->i2c;
+	uint32_t i2c_clock = 0U;
+
+	if (IS_ENABLED(STM32_I2C_DOMAIN_CLOCK_SUPPORT) && (cfg->pclk_len > 1)) {
+		if (clock_control_get_rate(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+					   (clock_control_subsys_t)&cfg->pclken[1],
+					   &i2c_clock) < 0) {
+			LOG_ERR("Failed call clock_control_get_rate(pclken[1])");
+			return -EIO;
+		}
+	} else {
+		if (clock_control_get_rate(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
+					   (clock_control_subsys_t) &cfg->pclken[0],
+					   &i2c_clock) < 0) {
+			LOG_ERR("Failed call clock_control_get_rate(pclken[0])");
+			return -EIO;
+		}
+	}
```

Comment:
Isn't i2c_clock already available in dev->config ?
If not , can it be ?

Reply:
That was the question : 
i2c_config_timing  is included in the struct i2c_stm32_config but as a constant that cannot be set during the `i2c_stm32_runtime_configure()`

Reply:
Ok, good point, I get it now. I need to think about it indeed.